src/argowrapper/routes/routes.py

```diff
@@ -149,6 +178,7 @@ def test():
 # submit argo workflow
 @router.post("/submit", status_code=HTTP_200_OK)
 @check_auth_and_team_project
+@check_payment_details
```

Comment:
continuing on the comment above: maybe the easiest solution is to just call this check_payment_details right before argo_engine.workflow_submission and pass the found billing_id to argo_engine.workflow_submission